pkg/apis/autoscaling/v1/conversion_test.go

```diff
@@ -0,0 +1,148 @@
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/kubernetes/pkg/apis/autoscaling"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	utilpointer "k8s.io/utils/pointer"
+)
+
+func TestConvert_autoscaling_HorizontalPodAutoscalerSpec_To_v1_HorizontalPodAutoscalerSpec(t *testing.T) {
+
+	type args struct {
+		in  *autoscaling.HorizontalPodAutoscalerSpec
+		out *autoscalingv1.HorizontalPodAutoscalerSpec
+		s   conversion.Scope
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			// Test case for #101370
+			"TestConversionWithCPUAverageValueAndUtilizationBoth",
+			args{
+				in: &autoscaling.HorizontalPodAutoscalerSpec{
+					MinReplicas: utilpointer.Int32Ptr(1),
+					MaxReplicas: 3,
+					Metrics: []autoscaling.MetricSpec{
+						{
+							Type: autoscaling.ResourceMetricSourceType,
+							Resource: &autoscaling.ResourceMetricSource{
+								Name: api.ResourceCPU,
+								Target: autoscaling.MetricTarget{
+									Type:         autoscaling.AverageValueMetricType,
+									AverageValue: resource.NewMilliQuantity(300, resource.DecimalSI),
+								},
+							},
+						},
+						{
+							Type: autoscaling.ResourceMetricSourceType,
+							Resource: &autoscaling.ResourceMetricSource{
+								Name: api.ResourceCPU,
+								Target: autoscaling.MetricTarget{
+									Type:               autoscaling.UtilizationMetricType,
+									AverageUtilization: utilpointer.Int32Ptr(70),
+								},
+							},
+						},
+					},
+				},
+				out: &autoscalingv1.HorizontalPodAutoscalerSpec{},
+				s:   nil,
+			},
+			false,
+		},
+		{
+			// Test case for #101370
+			"TestConversionWithCPUUtilizationOnly",
+			args{
+				in: &autoscaling.HorizontalPodAutoscalerSpec{
+					MinReplicas: utilpointer.Int32Ptr(1),
+					MaxReplicas: 3,
+					Metrics: []autoscaling.MetricSpec{
+						{
+							Type: autoscaling.ResourceMetricSourceType,
+							Resource: &autoscaling.ResourceMetricSource{
+								Name: api.ResourceCPU,
+								Target: autoscaling.MetricTarget{
+									Type:               autoscaling.UtilizationMetricType,
+									AverageUtilization: utilpointer.Int32Ptr(60),
+								},
+							},
+						},
+					},
+				},
+				out: &autoscalingv1.HorizontalPodAutoscalerSpec{},
+				s:   nil,
+			},
+			false,
+		},
+		{
+			"TestConversionWithCPUAverageValueOnly",
+			args{
+				in: &autoscaling.HorizontalPodAutoscalerSpec{
+					MinReplicas: utilpointer.Int32Ptr(1),
+					MaxReplicas: 3,
+					Metrics: []autoscaling.MetricSpec{
+						{
+							Type: autoscaling.ResourceMetricSourceType,
+							Resource: &autoscaling.ResourceMetricSource{
+								Name: api.ResourceCPU,
+								Target: autoscaling.MetricTarget{
+									Type:         autoscaling.AverageValueMetricType,
+									AverageValue: resource.NewMilliQuantity(200, resource.DecimalSI),
+								},
+							},
+						},
+					},
+				},
+				out: &autoscalingv1.HorizontalPodAutoscalerSpec{},
```

Comment:
I was expecting to see a couple of test cases where the output == utilization or average value (i.e. an example of where the user sets resource metrics that result in the output spec having a value set).  Those would be "things that worked before that keep working"

Reply:
I guess “TestConversionWithCPUUtilizationOnly” is the case you said that worked before and after.

Reply:
I guess “TestConversionWithCPUUtilizationOnly” is the case you said that worked before and after.

Reply:
I'm confused - I guess I expected one of the conversions to result in out being set (previously).  All of your examples show `out` as empty - I expected to see one that had a valid out.TargetCPUUtilizationPercentige set

Reply:
I probably understand your meaning and I will update the code later.
Thank you!